### PR TITLE
Terminal: Add Find next/previous key bindings. Fixes: #29661

### LINKED ISSUE #29661: No next and previous keyboard shortcuts for Find widget in Integrated Terminal
- VSCode Version: 1.14 Insiders
- OS Version: Windows 7

Steps to Reproduce:

1. Open Find widget and search for a string with several matches

There are no keyboard shortcuts for previous and next match (Shift + F3, F3).

### PR DESCRIPTION
**Pull request by cleidigh**
@Tyriar 
- Tried my best at naming nomenclature not perfect but matches most things
- Went through terminal services like focus and hide commands 
- Testing both Windows and OS X

Fixes: #29661

### CODE REVIEW

File: src/vs/workbench/parts/terminal/electron-browser/terminal.contribution.ts [hunk 1]
```diff
@@ -322,6 +324,14 @@ actionRegistry.registerWorkbenchAction(new SyncActionDescriptor(HideTerminalFind
 	primary: KeyCode.Escape,
 	secondary: [KeyCode.Shift | KeyCode.Escape]
 }, ContextKeyExpr.and(KEYBINDING_CONTEXT_TERMINAL_FOCUS, KEYBINDING_CONTEXT_TERMINAL_FIND_WIDGET_VISIBLE)), 'Terminal: Focus Find Widget', category);
+actionRegistry.registerWorkbenchAction(new SyncActionDescriptor(NextMatchTerminalFindWidgetAction, NextMatchTerminalFindWidgetAction.ID, NextMatchTerminalFindWidgetAction.LABEL, {
+	primary: KeyCode.F3
+}, KEYBINDING_CONTEXT_TERMINAL_FOCUS), 'Terminal: Find Next', category);
```
Comment (rebornix) [inline src/vs/workbench/parts/terminal/electron-browser/terminal.contribution.ts]: Usually for macOS, we use another keybinding -> https://github.com/Microsoft/vscode/blob/master/src/vs/editor/contrib/find/common/findController.ts#L372

File: src/vs/workbench/parts/terminal/electron-browser/terminal.contribution.ts [hunk 2]
```diff
@@ -322,6 +324,14 @@ actionRegistry.registerWorkbenchAction(new SyncActionDescriptor(HideTerminalFind
...
+actionRegistry.registerWorkbenchAction(new SyncActionDescriptor(PreviousMatchTerminalFindWidgetAction, PreviousMatchTerminalFindWidgetAction.ID, PreviousMatchTerminalFindWidgetAction.LABEL, {
+	primary: KeyMod.Shift | KeyCode.F3
+}, KEYBINDING_CONTEXT_TERMINAL_FOCUS), 'Terminal: Find Previous', category);
```
Comment (rebornix) [inline src/vs/workbench/parts/terminal/electron-browser/terminal.contribution.ts]: Same here.

**Comment by rebornix** [comment]
Just left a tiny comment about keybindings, it looks good from my side.

File: src/vs/workbench/parts/terminal/electron-browser/terminal.contribution.ts (hunk 1 above)
Comment (cleidigh) [inline src/vs/workbench/parts/terminal/electron-browser/terminal.contribution.ts]: @rebornix 
I looked at the OS X key bindings and Find Next appears to be F3 ^F3 for previous
is this not correct?

Comment (cleidigh) [inline src/vs/workbench/parts/terminal/electron-browser/terminal.contribution.ts]: Funny thing is that that's what I had first ;-)
will change back

Comment (rebornix) [inline src/vs/workbench/parts/terminal/electron-browser/terminal.contribution.ts]: Our builtin primary one is cmd+g but cmd+F3 works as well. Yeah let's make it consistent.

**Comment by cleidigh** [comment]
OS X Key bindings matched.

**Comment by cleidigh** [comment]
See new PR #32074

**Pull request closed without merging**